src/audio/host-zephyr.c

```diff
@@ -257,20 +257,18 @@ void host_common_update(struct host_data *hd, struct comp_dev *dev, uint32_t byt
 		ret = dma_buffer_copy_to(source, sink, hd->process, bytes);
 	}
 
-	/* assert dma_buffer_copy succeed */
-	if (ret < 0)
-		comp_err(dev, "host_common_update() dma buffer copy failed, dir %d bytes %d avail %d free %d",
-			 hd->ipc_host.direction, bytes,
+	buffer_release(sink);
+	buffer_release(source);
+
+	if (ret < 0) {
+		comp_err(dev, "host_common_update() copy failed, dir %d bytes %d avail %d free %d",
+			 dev->direction, bytes,
 			 audio_stream_get_avail_samples(&source->stream) *
 				audio_stream_frame_bytes(&source->stream),
 			 audio_stream_get_free_samples(&sink->stream) *
 				audio_stream_frame_bytes(&sink->stream));
```

Comment:
no, using `source` and `sink` after releasing them is wrong. We can postpone this until after the coherent API is removed from `struct comp_buffer`